modules/gapi/test/common/gapi_imgproc_tests_common.hpp

```diff
@@ -40,10 +139,59 @@ static void boundingRectTestBody(const In& in, const CompareRects& cmpF, cv::GCo
 {
     cv::Rect out_rect_gapi;
     boundingRectTestGAPI(in, std::move(args), out_rect_gapi);
-
     boundingRectTestOpenCVCompare(in, out_rect_gapi, cmpF);
 }
 
+//-------------------------------------------------------------------------------------------------
+
+template<typename In>
+static cv::GComputation fitLineTestGAPI(const In& in, const cv::DistanceTypes distType,
+                                        cv::GCompileArgs&& args, cv::Vec4f& out_vec_gapi)
+{
+    double paramDefault = 0., repsDefault = 0., aepsDefault = 0.;
+
+    cv::detail::g_type_of_t<In> g_in;
+    auto out = cv::gapi::fitLine2D(g_in, distType, paramDefault, repsDefault, aepsDefault);
+    cv::GComputation c(cv::GIn(g_in), cv::GOut(out));
+    c.apply(cv::gin(in), cv::gout(out_vec_gapi), std::move(args));
+    return c;
+}
+
+template<typename In>
+static cv::GComputation fitLineTestGAPI(const In& in, const cv::DistanceTypes distType,
+                                        cv::GCompileArgs&& args, cv::Vec6f& out_vec_gapi)
+{
+    double paramDefault = 0., repsDefault = 0., aepsDefault = 0.;
+
+    cv::detail::g_type_of_t<In> g_in;
+    auto out = cv::gapi::fitLine3D(g_in, distType, paramDefault, repsDefault, aepsDefault);
```

Comment:
Would it work if create `GOpaque` with an empty constructor for out? This `GOpaque` should be of `out_vec_gapi`'s type. And you can add `if` with type check and call appropriate `fitLine` function.

Reply:
No, you can't do that.
You mean,
```
cv::GOpaque<cv::Vec<float, dim>> out;
if (dim == 4)
    out = fitline2D;
else
    out = fitline3D;
```
fitline 2D/3D functions have different output types; and compiler doesn't care that you know you don't need to compile one of the cases - it will compile everything. And obviously will not be able to because there is type incompatibility in one of the cases

Reply:
Suggested another implementation. But ok.